metaflow/plugins/kfp/kfp.py

```diff
@@ -1130,15 +960,22 @@ def create_workflow_uid_op(s3_sensor_path: str):
                     "volume" in s.resource_requirements
                     for s in step_to_kfp_component_map.values()
                 ):
-                    workflow_uid_op = func_to_container_op(
-                        get_workflow_uid,
-                        base_image="gcr.io/cloud-builders/kubectl",
-                    )(
-                        work_flow_name="{{workflow.name}}",
-                        s3_sensor_path=s3_sensor_path,
-                    ).set_display_name(
-                        "get_workflow_uid"
-                    )
+                    get_workflow_uid_command = [
+                        "bash",
+                        "-ec",
+                        self._compile_time_bootstrap_cmd
+                        + (
+                            " && python -m metaflow.plugins.kfp.kfp_get_workflow_uid"
+                            f" --s3_sensor_path '{s3_sensor_path}'"
+                            " --workflow_name {{workflow.name}}"
+                        ),
+                    ]
+                    workflow_uid_op = dsl.ContainerOp(
+                        name="get_workflow_uid",
+                        image=self.base_image,
+                        command=get_workflow_uid_command,
+                        file_outputs={"Output": "/tmp/outputs/Output/data"},
+                    ).set_display_name("get_workflow_uid")
```

Comment:
same here, creating `workflow_uid_op` should be pulled out into it's own function 
creating `workflow_uid_op` have nothing todo with build_kfp_dag logic

Reply:
We are creating `workflow_uid_op` in a separate function already -> `self.create_workflow_uid_op()`. It is also outside the context of `build_kfp_dag()`. Am I missing something?

Reply:
Does it need to be in `create_kfp_pipeline_from_flow_graph` ??

Reply:
Same with `create_s3_sensor_op`

Reply:
Cool, `create_s3_sensor_op`, `create_workflow_uid_op`, `create_exit_handler_op`, `create_container_op`, all moved to the outer scope of the `KubeflowPipelines` class.